catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java

```diff
@@ -0,0 +1,124 @@
+ */
+package org.codice.alliance.imaging.chip.actionprovider;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+
+import org.codice.ddf.configuration.SystemBaseUrl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.io.ParseException;
+import com.vividsolutions.jts.io.WKTReader;
+
+import ddf.action.Action;
+import ddf.action.ActionProvider;
+import ddf.action.impl.ActionImpl;
+import ddf.catalog.data.Metacard;
+
+public class ImagingChipActionProvider implements ActionProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImagingChipActionProvider.class);
+
+    public static final String TITLE = "Chip Image";
+
+    public static final String DESCRIPTION =
+            "Opens a new window to enter the boundaries of an image chip for a Metacard.";
+
+    public static final String PATH = "/chipping/chipping.html";
+
+    public static final String ID = "image-chipping-action-view";
+
+    public static final String NITF_CONTENT_TYPE = "image/nitf";
+
+    @Override
+    public <T> List<Action> getActions(T input) {
+        if (input == null) {
+            LOGGER.warn("Metacard can not be null.");
+            return null;
+        }
+
+        if (canHandle(input)) {
+            final Metacard metacard = (Metacard) input;
+
+            URL url;
+            String chipPath = null;
+            try {
+                WKTReader wktReader = new WKTReader();
+                Geometry geometry = wktReader.read(metacard.getLocation());
+                String location = getBoxDimensionString(geometry.getEnvelope()
+                        .getCoordinates());
+                chipPath =
+                        SystemBaseUrl.getBaseUrl() + PATH + "?id=" + metacard.getId() + "&source="
+                                + metacard.getSourceId() + "&location=" + location.substring(0,
+                                location.length() - 1);
+                url = new URL(chipPath);
+            } catch (MalformedURLException e) {
+                LOGGER.debug("Invalid URL for chipping path : {}", chipPath, e);
+                return null;
+            } catch (ParseException e) {
+                LOGGER.debug("Cannot parse metacard geometry: {}", metacard.getLocation(), e);
+                return null;
+            }
+            return Arrays.asList(new ActionImpl(getId(), TITLE, DESCRIPTION, url));
+        }
+        return null;
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public <T> boolean canHandle(T subject) {
+        if (subject instanceof Metacard) {
+            Metacard metacard = (Metacard) subject;
+            String contentTypeName = metacard.getContentTypeName();
+            if (contentTypeName != null && contentTypeName.equals(NITF_CONTENT_TYPE)
```

Comment:
`contentTypeName != null && contentTypeName.equals(NITF_CONTENT_TYPE)` can be simplified to `NITF_CONTENT_TYPE.equals(contentTypeName)`.